### ISSUE TITLE
count() Low efficiency

### ISSUE THREAD

**Issue by Wizzercn** [balderdashy/sails-mysql#258]
SELECT COUNT(*) as count FROM (SELECT * FROM `sys_role` WHERE `sys_role`.`unitid` = 2 ) AS `sys_role`

why not : 

SELECT COUNT(*) as count  FROM `sys_role` WHERE `sys_role`.`unitid` = 2

**Comment by tjwebb** [comment]
PR would be appreciated.

**Comment by smanne** [comment]
any update on this, I am also effected with this issue?

**Comment by particlebanana** [comment]
Hey @Wizzercn sql queries are generated by https://github.com/balderdashy/waterline-sequel. Would you mind moving this question there so it can be tracked closer to the source?

**Comment by sailsbot** [comment]
Thanks for posting, @Wizzercn.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
